annotation-server/src/medications/medications.controller.ts

```diff
@@ -9,45 +14,23 @@ import { MedicationsService } from './medications.service';
 export class MedicationsController {
     constructor(private medicationsService: MedicationsService) {}
 
-    @ApiOperation({ summary: 'Fetch all medications with optional search' })
+    @ApiOperation({ summary: 'Fetch all medications' })
+    @ApiFindMedicationsQueries()
     @Get()
-    get(@Query() query: { search?: string }): Promise<Medication[]> {
-        if (query.search) {
-            return this.medicationsService.findMatchingMedications(
-                query.search,
-            );
-        }
-        return this.medicationsService.getAll();
+    async findAll(@Query() dto: FindMedicationQueryDto): Promise<Medication[]> {
+        return await this.medicationsService.findAll(
+            dto.limit ?? 0,
+            dto.offset ?? 0,
+            dto.search ?? '',
+            dto.sortby ?? 'name',
+            dto.orderby ?? 'asc',
+            dto.withGuidelines === 'true' ? true : false,
```

Comment:
I didn't find any good solution for this:
- Getters seem not to work on DTOs
- Extending native string by a toBoolean() function so that `'true'.toBoolean()` can be called doesn't seem to work
- Couldn't find a working implementation with class-transformer that transforms the string to bool immediately

We could make a helper function (exported by the .dto.ts file or directly declared in the med controller) but we would also call it 3 times in the controller (1 line each like it is right now).